### PR TITLE
feat(dom): DOM Element Assertion

### PR DESCRIPTION
**Pull request by ChristianSama**
Add `ElementAssertion` for the DOM plugin
- Added the `toBeInTheDocument` matcher.

### CODE REVIEW

**Review by fonsiher — COMMENTED** [review]
LGTM! Thank you Chris.

File: packages/dom/test/unit/lib/ElementAssertion.test.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        expect(test.toBeInTheDocument())
+          .toBeEqual(test);
+
+        expect(() => test.not.toBeInTheDocument())
+          .toThrowError(AssertionError)
+          .toHaveMessage("Expected the element to NOT be in the document");
+      });
+    });
+    context("when the element is not in the document", () => {
+      it("returns the assertion instance", () => {
```
Comment (matycarolina) [inline packages/dom/test/unit/lib/ElementAssertion.test.tsx:33]: hi @ChristianSama, we were wondering if maybe we should be a little more specific on the assertion instance being returned in each of the test cases? Or is it okay to leave it as is?
cc @JoseLion @fonsiher

File: packages/dom/test/unit/lib/ElementAssertion.test.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        expect(test.toBeInTheDocument())
+          .toBeEqual(test);
+
+        expect(() => test.not.toBeInTheDocument())
+          .toThrowError(AssertionError)
+          .toHaveMessage("Expected the element to NOT be in the document");
+      });
+    });
+    context("when the element is not in the document", () => {
+      it("Throws an assertion error", () => {
```
Comment (JoseLion) [inline packages/dom/test/unit/lib/ElementAssertion.test.tsx:33]:
```suggestion
      it("throws an assertion error", () => {
```

**Review by JoseLion — APPROVED** [review]
Looking great! I left a few minor suggestions, nothing blocking, though. Great job 🎉

**Review by matycarolina — APPROVED** [review]
LGTM 👍🏻

File: packages/dom/test/unit/lib/ElementAssertion.test.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      });
+    });
+    context("when the element is not in the document", () => {
+      it("returns the assertion instance", () => {
+        const detachedElement = document.createElement("div");
+
+        const test = new ElementAssertion(detachedElement);
+
+        expect(test.not.toBeInTheDocument())
+          .toBeEqual(test);
```
[the remark this answers is not in the document]
Comment (matycarolina) [reply packages/dom/test/unit/lib/ElementAssertion.test.tsx]: resolved in https://github.com/stackbuilders/assertive-ts/pull/129/commits/64c43ae6241f2e3e1e4b4ac13818abbeb6e7d4c9

File: packages/dom/src/lib/ElementAssertion.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      message: "Expected the element to be in the document",
+    });
+    const invertedError = new AssertionError({
+      actual: this.actual,
+      message: "Expected the element to NOT be in the document",
+    });
+
+    return this.execute({
+      assertWhen: (this.actual.ownerDocument.defaultView !== null
+        && this.actual.ownerDocument === this.actual.getRootNode({ composed: true })),
```
[the remark this answers is not in the document]
Comment (fonsiher) [reply packages/dom/src/lib/ElementAssertion.ts]: Resolved in https://github.com/stackbuilders/assertive-ts/pull/129/commits/64c43ae6241f2e3e1e4b4ac13818abbeb6e7d4c9

File: packages/dom/test/unit/lib/ElementAssertion.test.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import { ElementAssertion } from "../../../src/lib/ElementAssertion";
+
+function TestComponent(): ReactElement {
+  return (
+    <div>
+      <button>click me</button>
+    </div>
+  );
+}
+describe("[Unit] ElementAssertion.test.ts", () => {
```
[the remark this answers is not in the document]
Comment (fonsiher) [reply packages/dom/test/unit/lib/ElementAssertion.test.tsx]: Resolved in https://github.com/stackbuilders/assertive-ts/pull/129/commits/64c43ae6241f2e3e1e4b4ac13818abbeb6e7d4c9

File: packages/dom/test/unit/lib/ElementAssertion.test.tsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  describe(".toBeInTheDocument", () => {
+    context("when the element is in the document", () => {
+      it("returns the assertion instance", async () => {
+        const { findByRole } = render(<TestComponent />);
+        const button = await findByRole("button", {
+          name: "click me",
+        });
```
[the remark this answers is not in the document]
Comment (fonsiher) [reply packages/dom/test/unit/lib/ElementAssertion.test.tsx]: Resolved in https://github.com/stackbuilders/assertive-ts/pull/129/commits/64c43ae6241f2e3e1e4b4ac13818abbeb6e7d4c9

File: packages/dom/test/unit/lib/ElementAssertion.test.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+        expect(test.toBeInTheDocument())
+          .toBeEqual(test);
+
+        expect(() => test.not.toBeInTheDocument())
+          .toThrowError(AssertionError)
+          .toHaveMessage("Expected the element to NOT be in the document");
+      });
+    });
+    context("when the element is not in the document", () => {
```
[the remark this answers is not in the document]
Comment (fonsiher) [reply packages/dom/test/unit/lib/ElementAssertion.test.tsx]: Resolved in https://github.com/stackbuilders/assertive-ts/pull/129/commits/64c43ae6241f2e3e1e4b4ac13818abbeb6e7d4c9

File: packages/dom/test/unit/lib/ElementAssertion.test.tsx (hunk 2 above)
Comment (fonsiher) [reply packages/dom/test/unit/lib/ElementAssertion.test.tsx]: Resolved in https://github.com/stackbuilders/assertive-ts/pull/129/commits/64c43ae6241f2e3e1e4b4ac13818abbeb6e7d4c9

**Pull request merged**